src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts

```diff
@@ -87,12 +87,18 @@ export class ChallengeleaderboardComponent implements OnInit, AfterViewInit {
    * Currently selected phase split
    */
   selectedPhaseSplit: any = null;
+  selectedPhaseSplitLeaderboardToggle = false;
```

Comment:
Instead of setting this default value to `false`, you should get default value from the backend (what if challenge host changes the default value of this while creating challenge.)

Even how about if we use `show_leaderboard_by_latest_submission` directly from `selectedPhaseSplit` object defined in the typescript? and for which I think we need to update the serializer.

Reply:
1) I've set the default value to null. `!null` would evaluates to `true`.

![Screen Shot 2019-12-31 at 18 18 11](https://user-images.githubusercontent.com/34530026/71620012-eb22d100-2bf9-11ea-8af2-71084d3edea1.png)

I can't seem to see any boolean value regarding the default value of leaderboard sorting in the initialized API calls. Can you tell me where I can find the boolean value so when the API response is received, I can quickly reassign the value of `this.selectedPhaseSplitLeaderboardToggle`.

2) There is no `show_leaderboard_by_latest_submission` in any objects received. Can you tell me where I can find it?

And I can't find any serializer in the code.

---

In another note, are these changes necessary?

Reply:
Hey @drepram In order to get `show_leaderboard_by_latest_submission` from the `selectedPhaseSplit` object, you need to add this field in the serializer on EvalAI repo.

Reply:
@drepram Field already been added [here](https://github.com/Cloud-CV/EvalAI/pull/2568) so now you can get `show_leaderboard_by_latest_submission` field from the object.

Reply:
<img width="326" alt="Screen Shot 2020-01-02 at 23 57 21" src="https://user-images.githubusercontent.com/34530026/71680646-3afdc580-2dbd-11ea-9b40-4c8281744364.png">

Due to the non-existence of the `show_leaderboard_by_latest_submission` field in the `selectedPhaseSplit` object, I've decided to make a condition for it. Let me know if I need to make more changes, I would love to make them.
```javascript
if (this.phaseSplits[this.selectedPhaseSplit['id']].hasOwnProperty('show_leaderboard_by_latest_submission')) {
  BODY = JSON.stringify({
    'show_leaderboard_by_latest_submission': this.phaseSplits[this.selectedPhaseSplit['id']].show_leaderboard_by_latest_submission
  });
} else {
  BODY = JSON.stringify({
    'show_leaderboard_by_latest_submission': SELF.selectedPhaseSplitLeaderboardToggle
  });
}
```

Reply:
Please try rebuilding the django container then you'll be able to get that field from the `selectedPhaseSplit` object.